### ISSUE TITLE
Make it easier to use with other session extensions

### ISSUE THREAD

**Issue by vgavro** [Anorov/cloudflare-scrape#269]
There was related issue https://github.com/Anorov/cloudflare-scrape/issues/46 but it wasn't resolved properly, so I want to propose changes to make it easier to use with other session extensions (I'll create pull request if you consider this approach to be OK).
1. Move CloudflareScraper functionality to CloudflareScraperMixin, avoid direct subclass from Session in case we want to change mro.
2. Prepend `cf_` or `_cf_` or `_cfs_` to methods, properties and `__init__` arguments to avoid possible conflicts.
3. Use custom adapter in `__init__` only if it wasn't set already to allow custom adapter usage from other mixins, which may be subclassed from CloudflareAdapter if needed.
4. Use  `class CloudflareScraper(CloudflareScreperMixin, Session)` with deprecation warnings on old (renamed) args/kwargs and methods.

**Comment by pro-src** [comment]
Hi @vgavro,

Thanks for raising this issue. I have a couple of concerns with this approach.

Breaks backwards compatibility and a major version bump doesn't help much. If Cloudflare updates, users will be forced to update to the latest version. So there needs to be some commitment to backport updates while users migrate.

`requests` always mounts/sets an adapter for both HTTP and HTTPS. So the proposed check could be if the default adapters were used but I can't think of an easy way to check for it.

Besides that, I'm +1

**Comment by pro-src** [comment]
@vgavro 

Are you still considering this?

**Comment by vgavro** [comment]
Yes, I'll do my part on next week, please keep it open for a while)

**Comment by pro-src** [comment]
I'm still looking forward to seeing your work. Lmk, if there's an update.

**Comment by pro-src** [comment]
@vgavro Should this remain open?
